### PR TITLE
CLN/TST: Cleanup tests in test_ujson.py

### PR DESCRIPTION
**Pull request by gfyoung**
* Actually test the outputs
* Add more parameterization
* Take advantage of `pytest` features, such as exception contexts

This test module really needed an update 😄

### CODE REVIEW

File: pandas/tests/io/json/test_ujson.py [hunk 1]
```diff
@@ -40,116 +68,113 @@ def test_encodeDecimal(self):
...
         decoded = ujson.decode(encoded)
         assert sut == decoded
 
-    def test_encodeNonCLocale(self):
+    def test_encode_non_c_locale(self):
         import locale
-        savedlocale = locale.getlocale(locale.LC_NUMERIC)
+        saved_locale = locale.getlocale(locale.LC_NUMERIC)
+
         try:
```
Comment (jreback) [inline pandas/tests/io/json/test_ujson.py]: i think we have a contest manager for locale setting

Comment (gfyoung) [reply pandas/tests/io/json/test_ujson.py]: * We do, but it doesn't do the `Exception` checking that I think this test wants to do.
* I think I would want to tap into `_can_set_locale` instead.  Any reason why I shouldn't utilize into this private method?

Comment (jreback) [reply pandas/tests/io/json/test_ujson.py]: not familiar with that
but happy to use existing code (so u know why it’s private?)

**Comment by gfyoung** [comment]
@jreback : Everything is green.  PTAL.

**Comment by jreback** [comment]
thanks @gfyoung nice cleanups! keep em coming!

**Pull request merged**